apollos-church-api/src/data/ActionAlgorithm.js

```diff
@@ -70,7 +81,17 @@ class dataSource extends ActionAlgorithm.dataSource {
       .cache({ ttl: 60 })
       .get();
 
-    return filteredItems.map((item, i) => ({
+    // This filter ensures that the remaining items are False for shownOnHomePage and featuredOnHomePage
+    // There will be *at most* 3 items here that will need to be removed and
+    // the filteredItems array will be limited in size due to the above Rock filtering.
+    // This is reverse logic of the homeTabContentFeedAlgorithm above.
+    const nonPersonaOrHomePageItems = filteredItems.filter(
```

Comment:
why shouldn't this happen in the call to Rock? if you limit 3 and the first three are persona content, nothing will show up.

Reply:
There is not a limit of 3, this comment is just saying that this is not a resource intensive filter. At most, 1 feature item and 2 shown on home page items will be active and returned by the above Rock call. This filter removes any of those items (3 max) from the array.

We can't make this in the initial Rock call because you would need to compare the featuredonHomePage value and shownonHomePage value and confirm that they are False for the same content item. We could potentially have the initial request only check for shownonHomePage eq 'False'. Then we would need to do an additional filter here that would only remove any item that has featuredonHomePage eq 'True' (<--- This one would make it through the initial request because shownonHomePage would be False)

Reply:
Unless, I potentially do the filtering on `fliteredItems` instead of `combinedArrayByAttribute` and see if that works.

Reply:
I don't believe it will work there either. I tried to do something like `.filter(`((AttributeValues/ShownonHomePage/Value eq 'False) and (AttributeValues/FeaturedonHomePage/Value eq 'False'))`), but I can't expand AttributeValues on that request.

This was the error I was getting.
<img width="930" alt="Screen Shot 2021-11-24 at 1 47 31 PM" src="https://user-images.githubusercontent.com/72768221/143304157-138ebce3-3d62-49d5-9cc0-d65ea416e940.png">